src/python/serenity_types/pricing/derivatives/rates/yield_curve.py

```diff
@@ -0,0 +1,189 @@
+    """
+
+
+class RateSourceType(Enum):
+    """
+    Sources of rates & discount factors. In the most general case
+    the yield curve can be built from multiple imports, so we
+    tag each CurvePoint that is the input to the interpolated YC
+    with the particular source, e.g. we might have an 8H rate
+    at the short end from perpetual future funding rates and
+    implied forwards backed out from calendar spreads.
+    """
+
+    FUTURE_PX = "FUTURE_PX"
+    """
+    Implied forward backed out from observed future prices.
+    """
+
+    OPTION_PX = "OPTION_PX"
+    """
+    Implied forward backed out from observed option prices & spreads.
+    """
+
+    FIXING = "FIXING"
+    """
+    Observed traditional rate fixings.
+    """
+
+    FUNDING_RATE = "FUNDING_RATE"
+    """
+    Observed exchange perpetual future funding rate.
+    """
+
+    LENDING_RATE = "LENDING_RATE"
+    """
+    Observed CeFi / OTC or DeFi lending rate.
+    """
+
+    STAKING_RATE = "STAKING_RATE"
+    """
+    Observed proof-of-stake protocol staking rate.
+    """
+
+
+class InterpolationMethod(Enum):
+    """
+    Specific interpolation method used. Currently only supports flat-forward.
+    """
+    FLAT_FWD = "FLAT_FWD"
+
+
+class CurvePoint(BaseModel):
+    """
+    A discrete input point on the curve, with all the metadata describing
+    what is being provided and its source to help reproduce the results.
+    """
+
+    tenor: Optional[str]
+    """
+    A relative date code, e.g. 1Y or 3M.
+    """
+
+    pillar_date: Optional[date]
+    """
+    The specific forward date for the given rate and DF, e.g. the 1W point for today would be the next
+    business day a week ahead.
+    """
+
+    duration: Optional[float]
+    """
+    The duration for this point, expressed as a year fraction.
+    """
+
+    rate_source_type: RateSourceType
+    """
+    The type of input being provided for this CurvePoint, e.g. if it's from a 3M future,
+    this would be FUTURE_PX, while if it's from traditional rates fixings, it would be FIXINGS.
+    """
+
+    rate_sources: Optional[List[str]]
+    """
+    The specific rate sources used, e.g. OIS, SOFR or LIBOR; for LENDING_RATE, DeFi or other sources
+    used, e.g. CHAINLINK, IPOR or AAVE. For FUNDING_RATE this holds the UUID for the exchange Organization ID.
+    """
+
+    reference_assets: Optional[List[UUID]]
+    """
+    In the case where an implied forward is backed out from market observables, the assets observed.
+    """
+
+    mark_prices: Optional[List[float]]
+    """
+    In the case where a DF is backed out from the implied forward of a reference asset or basket thereof,
+    the observed prices that should go into the bootstrapping method.
+    """
+
+    rate: float
+    """
+    The input rate value, if DF not provided.
+    """
+
+    discount_factor: float
+    """
+    The input DF value, if rate not provided.
+    """
+
+
+class YieldCurve(BaseModel):
```

Comment:
Same as above, do we need YieldCurveDefinition?

```
class YieldCurveDefinition(BaseModel):
    """
    Basic metadata for yield curfe.
    """

    yield_curve_id: UUID
    """
    The unique identifier for this surface
    """

    display_name: str
    """
    Human-readable descrition of this curve, e.g. OIS (RAW) or OIS (Interpolated, FLAT_FWD)
    """

    start_datetime: datetime
    """
    UTC datetime for the earliest version in the database.
    """

    end_datetime: datetime
    """
    UTC datetime for the latest version in the database.
    """

```

Reply:
Yes, will do this one as well.

Reply:
Same as for VolatilitySurface, we can reuse YieldCurve here, no need for separate definition class.